### PR TITLE
[TPC-H] Dask query 15

### PR DESCRIPTION
**Pull request by milesgranger**
Part of https://github.com/coiled/dask-engineering/issues/234

This works for both dask.dataframe and dask-expr.

### CODE REVIEW

File: tests/tpch/test_dask.py [hunk 1]
```diff
@@ -306,3 +306,72 @@ def test_query_7(client, dataset_path, fs):
...
+        lineitem.groupby("l_suppkey")
+        .revenue.sum()
+        .to_frame()
+        .reset_index()
+        .rename(columns={"revenue": "total_revenue", "l_suppkey": "supplier_no"})
+    )
+
+    # Query
+    table = supplier.merge(
+        revenue, left_on="s_suppkey", right_on="supplier_no", how="left"
```
Comment (fjetter) [inline tests/tpch/test_dask.py:369]: I believe

```sql
from
        supplier,
        revenue
    where
        s_suppkey = supplier_no
```
is supposed to be an inner join

Comment (milesgranger) [reply tests/tpch/test_dask.py]: Fixed! Thanks! https://github.com/coiled/benchmarks/pull/1265/commits/9d984070a65d259087e70397ed07fb7c1bc23645

**Review by hendrikmakait — CHANGES_REQUESTED** [review]

**Comment by hendrikmakait** [comment]
CI failure for `test_query_15`: `ValueError: Not all divisions are known, can't align partitions. Please use `set_index` to set the index.`

Interestingly, this seems to work for a smaller scale factor (e.g., in `test_dask_results`)

**Comment by milesgranger** [comment]
@hendrikmakait, propose we xfail it then. Opened an issue https://github.com/dask-contrib/dask-expr/issues/812 in the meantime.

**Comment by phofl** [comment]
dask-expr should work now, the error is gone at least

**Review by hendrikmakait — APPROVED** [review]
Thanks, @milesgranger for implementing! @phofl, thanks for the fix, I've removed the `xfail` and the query seems to work.

**Pull request merged**